src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox/RichTextBox.cs

```diff
@@ -1824,6 +1823,58 @@ public unsafe int Find(string str, int start, int end, RichTextBoxFinds options)
     {
         ArgumentNullException.ThrowIfNull(str);
 
+        // Perform the find, will return ubyte position
+        int position = FindInternal(str, start, end, options);
+
+        // if we didn't find anything, or we don't have to select what was found,
+        // we're done
+        bool selectWord = (options & RichTextBoxFinds.NoHighlight) != RichTextBoxFinds.NoHighlight;
+        if (position != -1 && selectWord)
+        {
+            // Select the string found, this is done in ubyte units
+            CHARRANGE chrg = new()
+            {
+                cpMin = position
+            };
+
+            //Look for kashidas in the string.  A kashida is an arabic visual justification character
+            //that's not semantically meaningful.  Searching for ABC might find AB_C (where A,B, and C
+            //represent Arabic characters and _ represents a kashida).  We should highlight the text
+            //including the kashida.
+            const char kashida = (char)0x640;
+            ReadOnlySpan<char> kashidaString = [kashida];
+            int startIndex = FindInternal(kashidaString, position, position + str.Length, options);
+            if (startIndex == -1)
+            {
+                //No kashida in the string
```

Comment:
```suggestion
                // No kashida in the string
```
Ditto below with other comments in this method

Reply:
I've created a separate PR to enable the code analyzer for this: #10399